qcodes/loops.py

```diff
@@ -553,8 +553,15 @@ def _parameter_arrays(self, action):
             action_indices = ((),)
         else:
             raise ValueError('a gettable parameter must have .name or .names')
-
+        if hasattr(action, 'names') and hasattr(action, 'units'):
+            units = action.units
+        elif hasattr(action, 'unit'):
+            units = (action.unit,)
+        else:
+            units = tuple(['']*len(names))
         num_arrays = len(names)
+        num_units = len(units)
+        assert num_arrays == num_units
```

Comment:
Happy to get rid of it. In principle that should always be the case as units is constructed as 
`len(names)*" "` (wrapped in the right tuples) if not supplied. All of this extra handling was just to fix a test that explicitly deletes names from the loop and checks for the ValueError above